File: PHP/49.php
<?php

class Solution
{
    public function groupAnagrams(array $strs): array
    {
        $groups = [];

        foreach ($strs as $str) {
            $sortedStr = $this->sortString($str);
            $groups[$sortedStr][] = $str;
        }

        return array_values($groups);
    }

    private function sortString(string $str): string
    {
        $chars = str_split($str);

        sort($chars);

        return implode('', $chars);
    }
}

File: Csharp/Challenges/49.cs
namespace Csharp.Challenges;

public static partial class Solution
{
    public static IList<IList<string>> GroupAnagrams(string[] strs)
    {
        var groups = new Dictionary<string, IList<string>>();

        foreach (var str in strs)
        {
            var sorted = SortString(str);

            if (groups.ContainsKey(sorted))
                groups[sorted].Add(str);
            else
                groups[sorted] = new List<string>() { str };
        }

        return groups.Values.ToList();
    }

    private static string SortString(string str)
    {
        var chars = str.ToCharArray();

        Array.Sort(chars);

        return new string(chars);
    }
}
